soc/arm/silabs_exx32/common/soc.c

```diff
@@ -70,6 +71,13 @@ static ALWAYS_INLINE void clkInit(void)
 #endif
 }
 
+static ALWAYS_INLINE void dcDcInit(void) {
+#ifdef CONFIG_HAS_GECKO_DCDC
+	EMU_DCDCInit_TypeDef init = EMU_DCDCINIT_DEFAULT;
+	EMU_DCDCInit(&init);
```

Comment:
Don't you also need a change in `ext/hal/silabs/gecko/CMakeLists.txt` to build the DCDC driver?